chacra/controllers/binaries/__init__.py

```diff
@@ -128,6 +128,35 @@ def index_put(self):
         self.binary.path = self.save_file(file_obj)
         return dict()
 
+    @secure(basic_auth)
+    @index.when(method='DELETE', template='json')
+    def index_delete(self):
+        if not self.binary:
+            abort(404)
+        binary_path = self.binary.path
+        repo = self.binary.repo
+        project = self.binary.project
+        delete(self.binary)
+        try:
+            os.remove(binary_path)
+        except IOError:
+            # should we complain loudly here?
+            logger.exception("Could not remove the binary path: %s" % binary_path)
+        commit()
+        if repo.binaries.count() > 0:
+            # there are still binaries related to this repo, mark it to rebuild
+            repo.needs_update = True
+        else:
+            # there are no more binaries for this repo, delete the repo
+            delete(repo)
+
+        if project.binaries.count() == 0:
+            delete(project)
+
+        commit()
```

Comment:
this should be taken care of by the transactional hook, was that not the case? if so we should look into changing/updating the hook to take care of this automagically because it also takes care of rolling the transaction back nicely for us